### ISSUE TITLE
Confusion about MultiCatCategoricalPd

### ISSUE THREAD

**Issue by yangysc** [bowenliu16/rl_graph_generation#6]
Hello, I want to ask a problem about this special action space, and I hope I can get some help from here.

 In fact, in my own environment, I try to train an agent using ppo algorithm (without supervised learning in this paper), but I think t**he training fails because my action is similar to the paper** （**MultiCatCategoricalPd**）. I wonder   that if  an action A is composed of [A1, A2, A3], then another action B is [B1, B2, B3],  how to calculate the loss between them (**neglogp**, **kl**)?  In the code, it seems that we just treat each part of subaction as independent ones, but if the A1 is not the same with B1, there is no meaning to calcalute the similarity between A2 and B2. I mean **if the second subaction is conditioned on the first, how can we perform calculation without the need to care for the first subaction** .  

More importantly, my **second subaction space will be masked based on the first subaction**, so if A1 != B1, the valid space of A2 and B2 is not the same (I have to set the logits of invalid actions to -1e6), so c**alculating the neglogp between two actions under the same state will be huge if the first sub-action is different for these two actions**.   This issue has been around me about seveal months, so if you can help with this, I will appreciate that a lot.

Thanks in advance!
